support-workers/src/main/scala/com/gu/emailservices/DigitalPackEmailFields.scala

```diff
@@ -1,78 +1,242 @@
 package com.gu.emailservices
```

Comment:
this file was originally a single bulder with a couple of branch points to switch between corp/direct.
There is now a top level branching point between corp/direct/gift purchase/gift redemption which then calls out to a separate method for each one.
This is similar in some ways to the SendThankYouEmail buildEmail function, that has a top level branching point to choose between Contribution/DS/paper/weekly and calls out to each builder in a separate file.

I decided not to make a totally different file for each subcategory of DS, as there is some shared code, and putting it in a class rather than an object means we don't have to pass around so many parameters, as they are all in scope.  However I'm happy to keep things consistent in that way if people prefer that.

Reply:
Happy to keep it as is